datastore/gcloud/aio/datastore/array.py

```diff
@@ -0,0 +1,43 @@
+try:
+    from collections.abc import Sequence
+except ImportError:
+    from collections import Sequence
+from typing import Any
+from typing import Dict
+from typing import Iterator
+from typing import List
+
+import gcloud.aio.datastore.value
+
+
+# https://cloud.google.com/datastore/docs/reference/data/rest/v1/projects/runQuery#ArrayValue
+class Array(Sequence):
+
+    def __init__(self, items: List[Any]) -> None:
+        super(Array, self).__init__()
+        self.items = items
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Array):
+            return False
+        return self.items == other.items
+
+    def __repr__(self) -> str:
+        return repr(self.items)
```

Comment:
Could we define this as `return str(self.to_repr())` to match the `__repr__`s we have defined on our other classes?